apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -1,17 +1,36 @@
 defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
   alias Future.Code, as: Code
   alias Lexical.Ast
+  alias Lexical.Ast.Analysis
   alias Lexical.Document
   alias Lexical.Document.Location
   alias Lexical.Document.Position
+  alias Lexical.Formats
   alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
   alias Lexical.Text
 
   def definition(%Document{} = document, %Position{} = position) do
-    document
+    with {:ok, _, analysis} <- Document.Store.fetch(document.uri, :analysis),
+         {:ok, entity, _range} <- Entity.resolve(analysis, position) do
+      fetch_definition(entity, analysis, position)
+    end
+  end
+
+  defp fetch_definition({type, entity}, %Analysis{} = _analysis, %Position{} = _position)
+       when type in [:struct, :module] do
+    module = Formats.module(entity)
+
+    with [entry] <- Store.exact(module, type: type, subtype: :definition) do
```

Comment:
note: `Store.exact` can theoretically return multiple entries, which will be returned here instead of the ok/error tuple. It's unlikely to happen with modules, but there's nothing that states that you cannot name a module the same as another existing module on disk. 

prefer to swap this out with a case
```elixir
case Store.exact(module, type: type, subtype: :definition) do
  [entry] -> to_location(entry)
  _other -> {:error, :multiple_results}
end
```

Reply:
Theoretically speaking, such a situation could indeed occur.